main.tf

```diff
@@ -1,59 +1,165 @@
+  resource_allocation = var.resource_allocation
+
+  # Additions from old lambda sub-module
+  /*
+  dynamic "environment" {
+    for_each = length(var.environment) < 1 ? [] : [var.environment]
+    content {
+      variables = environment.value.variables
+    }
+  }
+
+  function_name                  = var.function_name
+  memory_size                    = var.memory_size
+  role                           = aws_iam_role.lambda.arn
+  source_code_hash               = filebase64sha256(var.filename)
+
+  dynamic "vpc_config" {
+    for_each = length(var.vpc_config) < 1 ? [] : [var.vpc_config]
+    content {
+      security_group_ids = vpc_config.value.security_group_ids
+      subnet_ids         = vpc_config.value.subnet_ids
+    }
+  }
+  */
+
+  # bonus points
+  #create_in_vpc     = var.create_in_vpc
+  #create_default_sg = var.create_default_sg
+  #enable_newrelic = var.enable_newrelic
+  #security_groups = concat(var.security_groups, tolist(aws_security_group.lambda_egress.id))
+}
+
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
 }
 
+resource "aws_iam_role" "lambda" {
+  name               = var.function_name
+  assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "cloudwatch_logs" {
+  role       = aws_iam_role.lambda.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "vpc_attachment" {
+  count = length(var.vpc_config) < 1 ? 0 : 1
+  role  = aws_iam_role.lambda.name
+
+  // see https://docs.aws.amazon.com/lambda/latest/dg/vpc.html
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole"
+}
+
+/*
 module "event-cloudwatch-scheduled-event" {
   source = "./modules/event/cloudwatch-scheduled-event"
   enable = lookup(var.event, "type", "") == "cloudwatch-scheduled-event" ? true : false
+  architecture = {}
 
   lambda_function_arn = module.lambda.arn
   schedule_expression = lookup(var.event, "schedule_expression", "")
 }
+*/
+
+module "lambda_cloudwatch_trigger" {
+  source  = "app.terraform.io/bankrate/lambda-cloudwatch-trigger/aws"
+  version = "~> 4.0"
+
+  lambda_function_arn = module.lambda.arn
+  schedule_expression = lookup(var.event, "schedule_expression", "")
+  environment         = var.environment
+  project             = var.project_name
+  owner               = var.owner
+}
+
+/*
+module "event-s3" {
+  source = "./modules/event/s3"
+  enable = lookup(var.event, "type", "") == "s3" ? true : false
+  architecture = {}
+
+  lambda_function_arn = module.lambda.arn
+  s3_bucket_arn       = lookup(var.event, "s3_bucket_arn", "")
+  s3_bucket_id        = lookup(var.event, "s3_bucket_id", "")
+}
+*/
 
+module "lambda_s3_trigger" {
+  source = "app.terraform.io/bankrate/lambda-s3-trigger/aws"
```

Comment:
```
  source = "app.terraform.io/bankrate/lambda-s3-trigger/aws"
  enable = lookup(var.event, "type", "") == "s3" ? true : false # something like this will do it, can be a struct similar to ecs-app's `architecture` flag structure. 
```